vignettes/report-writing-vignette.Rmd

```diff
@@ -0,0 +1,313 @@
+create the remaining three directories with the following code: 
+
+- `dir.create(here("data-raw"))`
+- `dir.create(here("data"))`
+- `dir.create(here("output"))`
+
+The R directory is used to store analysis scripts.
+
+The data-raw directory is used to store raw data files.
+
+The data directory is used to store cleaned or intermediate files.
+
+The output directory is used to store final output files that can be read into 
+the report.
+
+## Writing the Setup Chunk
+
+The Setup Chunk is where you set all of the default settings for your report. 
+In doing this, you want to create the Setup Chunk so that all of the necessary 
+settings can be set in a reproducible way. In other words, you want the code to 
+be able to run on any computer, not just yours.
+
+In the default template for R Markdown files, the Setup Chunk is the first code 
+chunk that looks like:
+
+```{r setup_chunk_example, eval = FALSE, echo = TRUE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+The first step in doing this is to load the needed packages. Bear in mind that 
+not all computers will have the the necessary packages installed already. Thus, 
+the code in the Setup Chunk needs to test whether these are already installed, 
+install them if they are not already installed, and then load the packages. 
+
+This can be done by running the following code:
+
+```{r package_install_loading, echo = TRUE, eval = FALSE}
+needed_packages <- c("ratlas", "tidyverse", "english", "kableExtra")
+load_packages <- function(x) {
+  if (!(x %in% installed.packages())) {
+    install.packages(x)
+  }
+  suppressPackageStartupMessages(require(x, character.only = TRUE))
+}
+vapply(needed_packages, load_packages, logical(1))
+
+extrafont::loadfonts(quiet = TRUE)
+```
+
+Additionally, the theme for ggplot should be specified. This can be done with 
+the either of the following code chunks.
+
+```{r set_ggplot_theme1, echo = TRUE, eval = FALSE}
+theme_atlas()
+```
+
+Or
+
+```{r set_ggplot_theme2, echo = TRUE, eval = FALSE}
+theme_atlas_ms()
+```
+
+Together, your first chunk of the file should look like:
+
+```{r setup_chunk, echo = TRUE, eval = FALSE}
+needed_packages <- c("ratlas", "tidyverse", "english", "kableExtra")
+load_packages <- function(x) {
+  if (!(x %in% installed.packages())) {
+    install.packages(x)
+  }
+  suppressPackageStartupMessages(require(x, character.only = TRUE))
+}
+vapply(needed_packages, load_packages, logical(1))
+
+system <- ifelse(Sys.info()["sysname"] == "Darwin", "Mac", "Windows")
+
+extrafont::loadfonts(quiet = TRUE)
+
+theme_atlas()
+#OR
+#theme_atlas_ms()
+```
+
+For simplicity, this setup code chunk can just be copy and pasted into the 
+rmarkdown file to write the report. Please note, however, that you should only
+select one of the two ATLAS themes (theme_atlas and theme_atlas_ms). 
+`Theme_atlas_ms()` has been commented out in the above code, but `theme_atlas()`
+be commented out if `theme_atlas_ms()` is desired.
+
+## Create and Cross Reference Tables
```

Comment:
I don't think we need a long descriptions for tables or figures. I would show a brief example of creating each, but link to external resources for details (e.g., the kableExtra package documentation, the R Markdown book for cross references, etc.).